subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -349,7 +349,14 @@ static uint8_t big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bi
 		iso_interval_us = iso_interval * PERIODIC_INT_UNIT_US;
 
 	} else {
-		lll_adv_iso->max_pdu = MIN(LL_BIS_OCTETS_TX_MAX, max_sdu);
+		if (framing) {
+			/* Try to allocate room for one SDU + header */
+			lll_adv_iso->max_pdu = MIN(LL_BIS_OCTETS_TX_MAX,
+						   max_sdu + PDU_ISO_SEG_HDR_SIZE +
+						    PDU_ISO_SEG_TIMEOFFSET_SIZE);
```

Comment:
i do not understand why bsim tests should failed? is this code that could cause it?

Reply:
I don't know why the test is failing now - it is caused by some of the (very) recent changes in main, it didn't fail a couple of days ago. I'll have to have a look at it when I have some free time

Reply:
Right - so I found the issue. There is a bug in the hosts bt_conn_tx_processor() implementation; If several connections are ready and one of them returns NULL in the tx_data_pull() callback it will not continue processing the rest - it requires another "data ready" trigger to happen first. This means that when the test application tries to send a payload on two BIS's at the same time we actually one receive the payloads for one of the BIS's in the controller - and we get two payloads at the same time instead. Since these two payloads have the same timestamp, and we now actually implement framed BIS, one will get dropped by ISO-AL.

I'll put up a separate PR to fix this, so we don't pollute this PR with host side fixes.

Trace excerpt:
```
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: bt_iso_chan_send: send-iso (no ts)[0m
d_00: @00:00:01.387051  conn_iso_send conn 0x84e8c80 buf 0x85787a8 has_ts 0
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: conn_iso_send: 0x85787a8 put on list[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_data_ready: DR[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_data_ready: already in list[0m
d_00: @00:00:01.387051  broadcast_sent_cb Sent payload with seq_num: 183 on stream 0x84a3980
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_notify: raise TX IRQ[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_complete_work: conn 0x84e8d5c[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_notify: conn 0x84e8d5c[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_notify: tx 0x84d6f60 cb 0x81a1d6b user_data (nil)[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_free: 0x84d6f60[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: bt_iso_chan_send: send-iso (no ts)[0m
d_00: @00:00:01.387051  conn_iso_send conn 0x84e8d5c buf 0x85787d0 has_ts 0
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: conn_iso_send: 0x85787d0 put on list[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_data_ready: DR[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_data_ready: raised[0m
d_00: @00:00:01.387051  broadcast_sent_cb Sent payload with seq_num: 183 on stream 0x84a39c4
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: tx_notify: raise TX IRQ[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: start[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: should_stop_tx: 0x84e8c80[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: processing conn 0x84e8c80[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: iso_data_pull: conn 0x84e8c80 amount 255[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: iso_data_pull: last frag, pop buf 0x8578780[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: pop: cb 0x81a1d6b userdata (nil)[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: TX process: conn 0x84e8c80 buf 0x8578780 (last)[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: conn 0x84e8c80 buf 0x8578780 len 102 buf->len 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: conn_tx_alloc: 0x84d6f78[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: move 0x8578780 ref in[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: get_data_frag: get-acl-frag: outside 0x8578780 window 0x857bbc0 size 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: send frag: buf 0x8578780 len 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: conn 0x84e8c80 buf 0x857bbc0 len 102 flags 0x02[0m
d_00: @00:00:01.387051  hci_iso_handle BIS payload for source_hdl 0 (HCI handle 3) with ts: 1377899/1377899
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: frag_destroy: [0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: start[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: should_stop_tx: 0x84e8c80[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: processing conn 0x84e8c80[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: iso_data_pull: conn 0x84e8c80 amount 255[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_iso: iso_data_pull: last frag, pop buf 0x85787a8[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: pop: cb 0x81a1d6b userdata (nil)[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: bt_conn_tx_processor: TX process: conn 0x84e8c80 buf 0x85787a8 (last)[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: conn 0x84e8c80 buf 0x85787a8 len 102 buf->len 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: conn_tx_alloc: 0x84d6f84[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: move 0x85787a8 ref in[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: get_data_frag: get-acl-frag: outside 0x85787a8 window 0x857bbc0 size 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: send frag: buf 0x85787a8 len 102[0m
d_00: @00:00:01.387051  [00:00:01.387,023] [0m<dbg> bt_conn: send_buf: conn 0x84e8c80 buf 0x857bbc0 len 102 flags 0x02[0m
d_00: @00:00:01.387051  hci_iso_handle BIS payload for source_hdl 0 (HCI handle 3) with ts: 1377899/1377899
```

Reply:
FYI looks like PR  #72090 is what broke the host; Looks like a complete rewrite of connection handling(!)

Reply:
Created issue #74321